src/device_tracker.cc

```diff
@@ -74,26 +74,34 @@ nlohmann::json TestResult::ToJson() const {
 DeviceTracker::DeviceTracker()
     : key_checker_(std::vector<std::vector<uint8_t>>()) {}
 
-void DeviceTracker::Initialize(const cbor::Value::ArrayValue& versions,
-                               const cbor::Value::ArrayValue& extensions,
-                               const cbor::Value::MapValue& options) {
+void DeviceTracker::Initialize(const cbor::Value::MapValue& info_map) {
   if (is_initialized_) {
     return;
   }
   is_initialized_ = true;
 
+  auto map_iter = info_map.find(CborValue(InfoMember::kVersions));
+  const cbor::Value::ArrayValue& versions = map_iter->second.GetArray();
```

Comment:
What if `map_iter == info_map.end()`? Should we assert that it's different?

Reply:
The returned CBOR is checked in `fido_commands.cc`, including all map keys and values. There are several functions that take vetted CBOR maps and read their content, assuming it is checked. The alternatives are:
- runtime errors on failure (the current solution, as `GetArray` internally checks)
- explicitly checking everywhere
- parsing the maps into dedicated data structures (OpenSK style)

Switching to the third option would require a lot of extra work at this point. I don't see the additional benefit of adding asserts other than documenting assumptions?

Reply:
1. I'm not sure that's a valid option. AFAICT `map_iter->second` should be undefined behavior if `map_iter == info_map.end()`.
2. Yes, this seems the less error-prone solution.
3. Yes, I also don't think it's worth it.
4. Have a precondition on the function that `info_map` contains an entry for "versions". Apparently that's the current option as I understand it. I'm fine with that if it's obvious that `info_map` should have a "versions" key. Otherwise I would document it in the header.

Reply:
True, I forgot about the `->second`, my answer only takes the `.GetArray` into account. I'll add the precondition to the function documentation and assert-check the `end()`.

Reply:
Done.